### ISSUE TITLE
OverflowError can escape `dateutil.parser.parse`

### ISSUE THREAD

**Issue by alex** [dateutil/dateutil#82]
`parse()` generally either returns a value, or raises a `ValueError`, however on some inputs it's possible for an `OverflowError` to escape:

```pycon
>>> dateutil.parser.parse("January 1st, 9999999999")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alex_gaynor/.virtualenvs/tempenv-000d93815348/lib/python2.7/site-packages/dateutil/parser.py", line 1008, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "/Users/alex_gaynor/.virtualenvs/tempenv-000d93815348/lib/python2.7/site-packages/dateutil/parser.py", line 404, in parse
    ret = default.replace(**repl)
OverflowError: signed integer is greater than maximum
```

(Found using the AFL fuzzer)

**Comment by pganssle** [comment]
This is a duplicate of issue #58. The resolution we decided upon then was to add `OverflowError` to the documentation as a possible failure mode, as this is the normal behavior of the datetime module. It's still not clear to me why datetime raises `OverflowError` when `OverflowError` would only be raised in situations where the number would raise `ValueError` *anyway*, so I'm open to catching this error and re-raising it as a `ValueError`, but on the other hand, I'm not totally comfortable disguising the source and type of errors raised in this context.

**Comment by jarondl** [comment]
@pganssle, you solved this by documenting the possibility in #63. If datetime does something weird we should report upstream, not disguise it, as you said.
